iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/Networks.java

```diff
@@ -92,7 +93,38 @@ public static void runScript(Network network, Reader reader, Writer out) {
         private double disconnectedShuntNegativeVolume = 0.0;
     }
 
-    public static void printBalanceSummary(String title, Network network, Logger logger) {
+    /**
+     * @deprecated Use {@link #printBalanceSummary(String, Network, Writer)} instead.
+     */
+    @Deprecated
+    public static void printBalanceSummary(String title, Network network, Logger logger) throws IOException {
+        try (OutputStream os = new OutputStream() {
+            private String mem = "";
+            private boolean closed = false;
+
+            @Override
+            public void write(int i) {
+                byte[] bytes = new byte[1];
+                bytes[0] = (byte) (i & 0xff);
+                mem = mem + new String(bytes);
```

Comment:
This is bad because it's quadratic no ?

Reply:
What is the purpose of this custom outputstream overriding?